launchpad/src/setup.rs

```diff
@@ -0,0 +1,207 @@
+elrond_wasm::imports!();
+
+#[elrond_wasm::module]
+pub trait SetupModule {
+    #[allow(clippy::too_many_arguments)]
+    #[init]
+    fn init(
+        &self,
+        launchpad_token_id: TokenIdentifier,
+        ticket_payment_token: TokenIdentifier,
+        ticket_price: Self::BigUint,
+        nr_winning_tickets: usize,
+        winner_selection_start_epoch: u64,
+        confirmation_period_in_epochs: u64,
+        claim_period_in_epochs: u64,
+    ) -> SCResult<()> {
+        require!(
+            launchpad_token_id.is_valid_esdt_identifier(),
+            "Invalid Launchpad token ID"
+        );
+        self.launchpad_token_id().set(&launchpad_token_id);
+
+        self.try_set_ticket_payment_token(&ticket_payment_token)?;
+        self.try_set_ticket_price(&ticket_price)?;
+        self.try_set_nr_winning_tickets(nr_winning_tickets)?;
+        self.try_set_winner_selection_start_epoch(winner_selection_start_epoch)?;
+        self.try_set_confirmation_period_in_epochs(confirmation_period_in_epochs)?;
+        self.try_set_claim_period_in_epochs(claim_period_in_epochs)?;
+
+        Ok(())
+    }
+
+    #[only_owner]
+    #[payable("*")]
+    #[endpoint(depositLaunchpadTokens)]
+    fn deposit_launchpad_tokens(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+    ) -> SCResult<()> {
+        let launchpad_token_id = self.launchpad_token_id().get();
+        require!(payment_token == launchpad_token_id, "Wrong token deposited");
+
```

Comment:
you may add a reuqire here of the quantity of the launchpad tokens to be equal number_of_winners of price_per_ticket